packages/gatsby-plugin-utils/src/polyfill-remote-file/utils/url-generator.ts

```diff
@@ -172,3 +175,75 @@ function generateImageArgs({
 
   return args.join(`&`)
 }
+
+export function generateImageUrlAlt(
+  source: {
+    url: string
+    filename: string
+    mimeType: string
+    internal: { contentDigest: string }
+  },
+  imageArgs: Parameters<typeof generateImageArgs>[0],
+  store?: Store
+): string {
+  const placeholderOrigin = `http://netlify.com`
+  const imageParams = generateImageArgsAlt(imageArgs)
+
+  const baseURL = new URL(`${placeholderOrigin}/${generateRoutePrefix(store)}`)
+
+  baseURL.search = imageParams.toString()
+  baseURL.searchParams.append(`url`, source.url)
+  baseURL.searchParams.append(`cd`, source.internal.contentDigest)
+
+  return `${baseURL.pathname}${baseURL.search}`
+}
+
+function generateRoutePrefix(store?: Store): string {
+  const state = store?.getState()
+
+  const pathPrefix = state?.program?.prefixPaths
+    ? state?.config?.pathPrefix
+    : ``
+
+  return pathPrefix + `.netlify/images`
```

Comment:
if pathPrefix is `/foo`, the resulting path would be `/foo/.netlify/images` - would this be correct / work? Doesn't `.netlify/X` need to be in root of "domain"?

Reply:
Ohh good point. Likely need to change that.